mozdownload/scraper.py

```diff
@@ -469,11 +479,14 @@ def get_build_info_for_date(self, date, build_index=None):
         self.logger.info('Retrieving list of builds from %s' % url)
         parser = DirectoryParser(url, authentication=self.authentication,
                                  timeout=self.timeout_network)
-        regex = r'%(DATE)s-(\d+-)+%(BRANCH)s%(L10N)s$' % {
+        is_mobile = self.application == 'mobile'
```

Comment:
I would move it inline to line 488 and also make use of `not in ('mobile')`, so default comes first as for l10n selection.